src/engine/LocalVocab.cpp

```diff
@@ -10,47 +11,48 @@
 
 // _____________________________________________________________________________
 LocalVocab LocalVocab::clone() const {
-  LocalVocab localVocabClone;
-  localVocabClone.otherWordSets_ = otherWordSets_;
-  localVocabClone.otherWordSets_.push_back(primaryWordSet_);
-  // Return the clone.
-  return localVocabClone;
+  LocalVocab result;
+  result.mergeWith(std::span{this, 1});
+  AD_CORRECTNESS_CHECK(result.size_ == size_);
+  return result;
 }
 
 // _____________________________________________________________________________
 LocalVocab LocalVocab::merge(std::span<const LocalVocab*> vocabs) {
-  LocalVocab res;
-  res.mergeWith(vocabs |
-                std::views::transform(
-                    [](const LocalVocab* localVocab) -> const LocalVocab& {
-                      return *localVocab;
-                    }));
-  return res;
+  LocalVocab result;
+  result.mergeWith(vocabs |
+                   std::views::transform(
+                       [](const LocalVocab* localVocab) -> const LocalVocab& {
+                         return *localVocab;
+                       }));
```

Comment:
If you want to, you can add the correpsonding functor to the end of  `util/TransparentFunctors.h`, s.t. you can write here
`vocabs | std::views::transform(ad_utility::dereference)`.